crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -543,20 +553,40 @@ fn execute_syscall(
             Ok(())
         }
         DeprecatedSyscallSelector::Deploy => Err(HintError::CustomHint(Box::from(
-            "Use snforge_std::ContractClass::deploy instead of deploy_syscall"
-                .to_string(),
+            "Use snforge_std::ContractClass::deploy instead of deploy_syscall".to_string(),
         ))),
         DeprecatedSyscallSelector::ReplaceClass => Err(HintError::CustomHint(Box::from(
-            "Replace class can't be used in tests"
-                .to_string(),
+            "Replace class can't be used in tests".to_string(),
         ))),
-        DeprecatedSyscallSelector::GetBlockHash => Err(HintError::CustomHint(Box::from(
-            "Get block hash is temporarily disabled in tests: https://github.com/foundry-rs/starknet-foundry/issues/686"
-                .to_string(),
-        ))),        _ => blockifier_syscall_handler.execute_hint(vm, exec_scopes, hint_data, constants),
+        DeprecatedSyscallSelector::GetBlockHash => {
+            execute_get_block_hash(vm, &mut system_ptr.clone(), cheatnet_state)?;
+            Ok(())
+        }
+        _ => blockifier_syscall_handler.execute_hint(vm, exec_scopes, hint_data, constants),
     }
 }
 
+fn execute_get_block_hash(
+    vm: &mut VirtualMachine,
+    system_ptr: &mut Relocatable,
+    _cheatnet_state: &CheatnetState,
+) -> Result<(), HintError> {
+    let _selector = stark_felt_from_ptr(vm, system_ptr)?;
+    let SyscallRequestWrapper {
+        gas_counter,
+        request: _,
+    } = SyscallRequestWrapper::<GetBlockHashRequest>::read(vm, system_ptr)?;
+
+    let sc_response = SyscallResponseWrapper::Success {
+        gas_counter: gas_counter - GET_BLOCK_HASH_GAS_COST,
+        response: GetBlockHashResponse {
+            block_hash: BlockHash(StarkFelt::from(0_u32)),
```

Comment:
Why block_hash is always 0? 👀

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/725#discussion_r1338419739